src/Symfony/Component/AccessToken/Bridge/OAuth/ClientCredentialsProvider.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AccessToken\Bridge\OAuth;
+
+use Symfony\Component\AccessToken\AccessToken;
+use Symfony\Component\AccessToken\AccessTokenInterface;
+use Symfony\Component\AccessToken\Bridge\AbstractProvider;
+use Symfony\Component\AccessToken\CredentialsInterface;
+use Symfony\Component\AccessToken\Exception\ProviderFetchException;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * OAuth2 "client_credentials" grant connector.
+ *
+ * This provider implements strictly OAuth 2.0 standard, and should work with all
+ * remote services that implement it correctly.
+ *
+ * "client_credentials" workflow may use the "refresh_token" for refreshing tokens
+ * but this is not implemented yet,
+ *
+ * This implementation was tested with:
+ *   - Microsoft Azure OAuth implementation v2.0
+ *
+ * @todo Test with Orange
+ * @todo Test with Google
+ * @todo Test with Salesforce
+ * @todo Test with ... ?
```

Comment:
Can you move these todos to the PR description?

Reply:
I have a strong opinion about this: and no, I won't remove them, because it's here to trigger some interest to reviewers, in order to trigger discussion not about code conventions, but about design itself, how much should we test ? What is needed to validate this as a prototype ? Those comments, of course, will not make it into Symfony core if the PR is accepted, but those are the very last one I'll remove at the very end, because their meaning is all about this PR, it's why I do it, and what needs to be discussed.

So, I agree with you, but no, I wont remove them until the usefulness of this bundle is stated.

Reply:
Ok